src/libraries/System.Text.RegularExpressions/gen/UpgradeToRegexGeneratorAnalyzer.cs

```diff
@@ -285,7 +285,7 @@ private static bool ProjectSupportsRegexSourceGenerator(Compilation compilation,
                 return false;
             }
 
-            if (compilation.SyntaxTrees.FirstOrDefault().Options is CSharpParseOptions options && options.LanguageVersion <= (LanguageVersion)1000)
+            if (((CSharpCompilation)compilation).LanguageVersion <= (LanguageVersion)1000)
```

Comment:
While I expect this to work in our cases (since we only support CSharp with this analyzer) doing the direct cast is a potential for a NRE, which is why I had the cast deconstructed. I don't really have a strong opinion for it though, could go either way.

Reply:
The cast is safe per the attribute `[DiagnosticAnalyzer(LanguageNames.CSharp)]`. I think direct casts like this are used a lot in both dotnet/roslyn and dotnet/roslyn-analyzers.